lib/dns.js

```diff
@@ -153,7 +153,9 @@ Client.prototype.destroy = function() {
     var cb = args[args.length - 1];
     args[args.length - 1] = function(err, result) {
       client.destroy();
-      cb(null, result);
+      
+      if (err) cb(err);
```

Comment:
Let's just do it in a following way: `cb(err, result)`. It is quite similar, IMHO.